wp-includes/components/class-wp-service-worker-configuration-component.php

```diff
@@ -54,20 +54,26 @@ public function get_script() {
 		$current_scope = wp_service_workers()->get_current_scope();
 		$workbox_dir   = 'wp-includes/js/workbox/';
 
-		if ( WP_DEBUG ) {
+		$script = '';
+		if ( SCRIPT_DEBUG ) {
+			$enable_debug_log = defined( 'WP_SERVICE_WORKER_DEBUG_LOG' ) && WP_SERVICE_WORKER_DEBUG_LOG;
+			if ( ! $enable_debug_log ) {
+				$script .= "self.__WB_DISABLE_DEV_LOGS = true;\n";
+			}
+
 			// Load with importScripts() so that source map is available.
-			$script = sprintf(
+			$script .= sprintf(
 				"importScripts( %s );\n",
 				wp_service_worker_json_encode( PWA_PLUGIN_URL . $workbox_dir . 'workbox-sw.js' )
 			);
 		} else {
 			// Inline the workbox-sw.js to avoid an additional HTTP request.
-			$script = file_get_contents( PWA_PLUGIN_DIR . '/' . $workbox_dir . 'workbox-sw.js' ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
-			$script = preg_replace( '://# sourceMappingURL=.+?\.map:', '', $script );
+			$script .= file_get_contents( PWA_PLUGIN_DIR . '/' . $workbox_dir . 'workbox-sw.js' ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+			$script .= preg_replace( '://# sourceMappingURL=.+?\.map:', '', $script );
```

Comment:
This probably shouldn't be appended - likely an oversight :)

Reply:
Good catch. Fixed in 22605b45123c3dcbb43b610b50a1b0a7e572ec32.